src/coreclr/vm/comdelegate.cpp

```diff
@@ -1725,8 +1729,10 @@ extern "C" void QCALLTYPE Delegate_Construct(QCall::ObjectHandleOnStack _this, Q
     if (COMDelegate::NeedsWrapperDelegate(pMeth))
         refThis = COMDelegate::CreateWrapperDelegate(refThis, pMeth);
 
+    refThis->SetMethodDesc(pMethOrig);
```

Comment:
I'm not 100% positive I use the correct descs when setting the field from VM in places like these.

Reply:
Is this trying to be a perf fix or correctness fix?

`Delegate_Construct` is a rare slow path. I do not think we need to bother optimizing it.

Reply:
> Is this trying to be a perf fix or correctness fix?
> 
> `Delegate_Construct` is a rare slow path. I do not think we need to bother optimizing it.

It was supposed to be a perf improvement, but as per discussion with @AndyAyersMS, if we could guarantee the desc is always reliably set by the VM when a delegate is created and never lazily initialized, delegate GDV could then be improved by switching to it. I feel like guaranteeing that should be left to a future PR though since it needs somebody with more VM/reflection knowledge to check all the possible paths.

Reply:
I would leave it out from this PR. It is hard to see whether these places use the right MethodDesc in all corner cases. 

Also, it is very incomplete, so it won't enable the JIT to do anything interesting with the MethodDesc either.